docs/modules/ROOT/pages/reference/concepts.adoc

```diff
@@ -45,6 +45,12 @@ inventory to install the software.
 Components can optionally provide Jsonnet libraries which can be used by other components.
 To allow Commodore to make component libraries available to other components, they must be placed in directory `lib/`.
 
+Components can specify Jsonnet dependencies in a `jsonnetfile.json` in the component's root directory.
+Commodore uses https://github.com/jsonnet-bundler/jsonnet-bundler[jsonnet-bundler] to fetch all components' dependencies.
+
+Optionally, components can provide a `jsonnetfile.jsonnet` instead which will be rendered to JSON before Commodore fetches component dependencies.
+In this case, external variables which are used in the `jsonnetfile.jsonnet` must be provided in the inventory key `parameters.<component_name>.jsonnetfile_parameters`.
```

Comment:
What is the reason behind restricting to a hard coded parameter key?

Reply:
Good point, I also don't see a reason to restrict this?

Reply:
Jsonnet expects a flat dict of strings to strings for the external variables. Therefore we cannot directly pass in the component's parameters dict, as that dict can contain any values.